sample/rules_test.go

```diff
@@ -2500,7 +2500,232 @@ func TestRulesRootSpanContext(t *testing.T) {
 			},
 			ExpectedKeep: false,
 			ExpectedRate: 10,
-			ExpectedName: "root doesnt match, next span doesnt match, third span matches",
+			ExpectedName: "root doesn't match, next span doesn't match, third span matches",
+		},
+		{
+			Rules: &config.RulesBasedSamplerConfig{
+				Rules: []*config.RulesBasedSamplerRule{
+					{
+						Name:       "no root span",
+						SampleRate: 10,
+						Conditions: []*config.RulesBasedSamplerCondition{
+							{
+								Fields:   []string{"root.http.status_code"},
+								Operator: config.EQ,
+								Value:    "500",
+							},
+						},
+					},
+				},
+			},
+			Spans: []*types.Span{
+				{
+					Event: types.Event{
+						Data: map[string]interface{}{
+							"http.status_code": "200",
+						},
+					},
+				},
+				{
+					Event: types.Event{
+						Data: map[string]interface{}{
+							"http.status_code": "500",
+						},
+					},
+				},
+			},
+			ExpectedKeep: true,
+			ExpectedRate: 1,
+			ExpectedName: "no rule matched",
+		},
+		{
+			Rules: &config.RulesBasedSamplerConfig{
+				Rules: []*config.RulesBasedSamplerRule{
+					{
+						Name:       "no root span with multiple fields",
+						SampleRate: 10,
+						Conditions: []*config.RulesBasedSamplerCondition{
+							{
+								Fields:   []string{"root.foo", "foo"},
+								Operator: config.EQ,
+								Value:    100.01,
+							},
+						},
+					},
+				},
+			},
+			Spans: []*types.Span{
+				{
+					Event: types.Event{
+						Data: map[string]interface{}{
+							"foo": 100.01,
+						},
+					},
+				},
+				{
+					Event: types.Event{
+						Data: map[string]interface{}{
+							"foo": 99.1,
+							"bar": 100.10,
+						},
+					},
+				},
+			},
+			ExpectedKeep: false,
+			ExpectedRate: 10,
+			ExpectedName: "no root span with multiple fields",
+		},
+		{
+			Rules: &config.RulesBasedSamplerConfig{
+				Rules: []*config.RulesBasedSamplerRule{
+					{
+						Name:       "no root span, checking other spans in trace",
+						SampleRate: 10,
+						Conditions: []*config.RulesBasedSamplerCondition{
+							{
+								Fields:   []string{"root.foo", "foo"},
+								Operator: config.EQ,
+								Value:    true,
+							},
+						},
+					},
+				},
+			},
+			Spans: []*types.Span{
+				{
+					Event: types.Event{
+						Data: map[string]interface{}{
+							"foo": true,
+						},
+					},
+				},
+				{
+					Event: types.Event{
+						Data: map[string]interface{}{
+							"foo": true,
+						},
+					},
+				},
+			},
+			ExpectedKeep: false,
+			ExpectedRate: 10,
+			ExpectedName: "no root span, checking other spans in trace",
+		},
+		{
+			Rules: &config.RulesBasedSamplerConfig{
+				Rules: []*config.RulesBasedSamplerRule{
+					{
+						Name:       "NotExists🦶🔫/root span exists, rule uses has-root-span guard, field is missing",
```

Comment:
Added emoji for fun.

Added the slash to sub-categorize these tests in the test suite run output:

In a fancy terminal:
![Screenshot 2024-02-23 at 9 47 35 AM](https://github.com/honeycombio/refinery/assets/517302/82bc44a3-1705-4733-87f1-0fa543823ba3)

In VS Code's test runner:
![Screenshot 2024-02-23 at 9 48 23 AM](https://github.com/honeycombio/refinery/assets/517302/082a9319-f849-4192-b364-da4cbfe061ee)

Reply:
This is a good add. Appreciate the categorization as well.